### PR TITLE
Add custom image during loading of the geppetto application

### PR DESCRIPTION
**Pull request by ddelpiano**
- Custom image can be placed in the webapp/images/splash.png
- the custom image will replace the text "LOADING... [~3MB]" and displayed instead of it during the load.

This is just an extension of Jesus' branch but based on the latest development, so I opened a new branch from my local development.

### CODE REVIEW

**Comment by ddelpiano** [comment]
@Robbie1977 could you review the changes in the stackviewer please?
I had some conflict and resolved them but a second look will make me feel in peace, thx.

@gidili could you review the geppetto.ejs and if Robbie is fine with the stackviewer changes merge this please? Thx!

**Comment by ddelpiano** [comment]
address [this](https://github.com/VirtualFlyBrain/geppetto-vfb/issues/467)

File: js/components/widgets/stackViewer/StackViewerComponent.js [hunk 1]
```diff
@@ -443,18 +443,21 @@ define(function (require) {
...
-                      if (i !== 0 || index !== 0) { // don't select template
-                        if (index == 0) {
-                          if (!shift) {
-                            that.state.objects.push(that.state.label[i]);
-                          }
-                        } else {
-                          if (typeof that.props.templateDomainIds !== 'undefined' && typeof that.props.templateDomainNames !== 'undefined' && typeof that.props.templateDomainIds[index] !== 'undefined' && typeof that.props.templateDomainNames[index] !== 'undefined' && that.props.templateDomainNames[index] !== null) {
-                            that.state.objects.push(that.props.templateDomainNames[index]);
-                            break;
+                  var currentPosition = that.renderer.plugins.interaction.mouse.getLocalPosition(that.stack);
```
Comment (filippomc) [inline js/components/widgets/stackViewer/StackViewerComponent.js]: This doesn't seem related to the current scope of the pr. Is this here for a reason or it's just coming form local merging?

File: js/pages/geppetto/geppetto.ejs [hunk 2]
```diff
@@ -45,7 +45,10 @@ $content
 
 
 <div id="mainContainer">
-	<div id="loadingText" style="text-align:center; font-family: consolas; margin-top: 50px;} ">LOADING... [~3MB]</div>
+	<div id="loadingText" style="display:grid; height: '100%'; text-align:center; font-family: consolas;">
+		<img src="geppetto/build/splash.png" style="max-width: 101%;max-height: 100; margin-top: -9px;margin-left: -8px;" alt="" onerror='document.getElementById("loadingMessage").style.display="block"' >
```
Comment (filippomc) [inline js/pages/geppetto/geppetto.ejs]: @ddelpiano just two doubts:
- The negative margins and max-width 101% don't look very clean, seems something related to a specific image we re loading.
- Where the image `geppetto/build/splash.png` comes from? Do we have to place directly into the build directory or there something in webpack that is moving this file there?

File: js/components/widgets/stackViewer/StackViewerComponent.js (hunk 1 above)
Comment (ddelpiano) [reply js/components/widgets/stackViewer/StackViewerComponent.js]: local merge as you pointed out, as written in the previous message I added robbie as reviewer to check this is fine since he was working with this.

File: js/pages/geppetto/geppetto.ejs (hunk 2 above)
Comment (ddelpiano) [reply js/pages/geppetto/geppetto.ejs]: fixing this.

File: js/components/widgets/stackViewer/StackViewerComponent.js (hunk 1 above)
Comment (filippomc) [reply js/components/widgets/stackViewer/StackViewerComponent.js]: Can be that this is also part of another pr? This always happens to me also when I'm working on a project and yet I have to find a clean way to get rid of additional code. Doing a rebase on the remote dev could be a solution but the history is stays littered if there was a push. My best solution for now is to start a new feature branch from the last dev and add only the code I need there for the pr.
